CMakeLists.txt

```diff
@@ -483,6 +483,15 @@ if(LINUX)
   add_definitions(-D__linux__)
 endif(LINUX)
 
+# Profiling build
+option(WITH_PROFILING "build with profiling build" OFF)
+
+if(WITH_PROFILING)
```

Comment:
we can always pass the `CMAKE_CXX_FLAGS` and `CMAKE_C_FLAGS` at the configure-time or build-time. like:

```
cmake -DCMAKE_CXX_FLAGS="-fno-omit-frame-pointer -O2" ..
```

or 
```
make vstart-base CMAKE_CXX_FLAGS="-fno-omit-frame-pointer -O2"
```

Reply:
since compile-time `CMAKE_CXX_FLAGS` and `CMAKE_C_FLAGS` offer us much more flexibility. probably #17358 can help more developer to get to know this.